aptos-move/framework/supra-framework/sources/automation_registry.move

```diff
@@ -245,69 +155,43 @@ module supra_framework::automation_registry {
         payload_tx: vector<u8>,
         expiry_time: u64,
         max_gas_amount: u64,
-        gas_price_cap: u64
+        gas_price_cap: u64,
+        parent_hash: vector<u8>
     ) acquires AutomationRegistry {
         let registry_data = borrow_global_mut<AutomationRegistry>(@supra_framework);
 
-        // todo : well formedness check of payload_tx
+        //Well formedness check of payload_tx is done in native layer beforehand.
 
         let current_time = timestamp::now_seconds();
-        assert!(expiry_time > current_time, EINVALID_EXPIRY_TIME);
-
-        let expiry_time_duration = expiry_time - current_time;
-        assert!(expiry_time_duration < registry_data.duration_upper_limit, EEXPIRY_TIME_UPPER);
+        let task_duration = expiry_time - current_time;
```

Comment:
subtraction is not safe

Reply:
True, I have moved check to registry_state and did not take this into account.